cmd/provider/main.go

```diff
@@ -144,6 +148,23 @@ func main() {
 
 	log := logging.NewLogrLogger(zl.WithName("provider-ceph"))
 
+	// Init otel tracer provider
+	tp, err := traces.InitTracerProvider(*metricsExportAddress, *metricsExportTimeout, *metricsExportInterval)
+	if err != nil {
+		log.Debug("error starting tracer provider", "error", err.Error())
+	}
+	if tp != nil {
+		defer func() {
+			ctx, cancel := context.WithTimeout(context.Background(), *metricsExportTimeout)
```

Comment:
It looks dead code for me, because we don't have proper graceful termination. Fix me if i'm wrong.

Reply:
FYI this defer func is flushing the remaining traces stored in-memory and sending them to the server.

Even though we may not have adopted a proper approach to handle the graceful shutdown of the process and all its moving parts, not all k8s operations will forcefully try to terminate it. Some give it time to shutdown properly and so this is not dead code so IMO we would benefit from including it.

Reply:
Discussed offline. Due to the default behavior of `kingpin.FatalIfError()` we would call `os.Exit()` on any errors and thus never execute the defer function mentioned above.

This was sorted out by overwriting the function called by `kingpin.FatalIfError()` through a native method offered by the lib so that it flushes the traces just before making the `os.Exit()` call.